ldk-server/src/api/error.rs

```diff
@@ -52,3 +53,67 @@ impl fmt::Display for LdkServerErrorCode {
 		}
 	}
 }
+
+impl From<NodeError> for LdkServerError {
+	fn from(error: NodeError) -> Self {
+		let (message, error_code) = match error {
+			NodeError::InvalidAddress
+			| NodeError::InvalidSocketAddress
+			| NodeError::InvalidPublicKey
+			| NodeError::InvalidSecretKey
+			| NodeError::InvalidOfferId
+			| NodeError::InvalidNodeId
+			| NodeError::InvalidPaymentId
+			| NodeError::InvalidPaymentHash
+			| NodeError::InvalidPaymentPreimage
+			| NodeError::InvalidPaymentSecret
+			| NodeError::InvalidAmount
+			| NodeError::InvalidInvoice
+			| NodeError::InvalidOffer
+			| NodeError::InvalidRefund
+			| NodeError::InvalidChannelId
+			| NodeError::InvalidNetwork
+			| NodeError::InvalidUri
+			| NodeError::InvalidQuantity
+			| NodeError::InvalidNodeAlias
+			| NodeError::InvalidDateTime
+			| NodeError::InvalidFeeRate
+			| NodeError::UriParameterParsingFailed => {
+				(error.to_string(), LdkServerErrorCode::InvalidRequestError)
+			},
+
+			NodeError::ConnectionFailed
+			| NodeError::InvoiceCreationFailed
+			| NodeError::InvoiceRequestCreationFailed
+			| NodeError::OfferCreationFailed
+			| NodeError::RefundCreationFailed
+			| NodeError::PaymentSendingFailed
+			| NodeError::InvalidCustomTlvs
+			| NodeError::ProbeSendingFailed
+			| NodeError::ChannelCreationFailed
+			| NodeError::ChannelClosingFailed
+			| NodeError::ChannelConfigUpdateFailed
+			| NodeError::DuplicatePayment
```

Comment:
Haven't looked through these exhaustively, but for BOLT11 `DuplicatePayment` is returned if the user tries to pay an invoice with the same payment hash twice. So shouldn't it map to an `InvalidRequestError`? Though for BOLT12 it would happen if we randomly generate the same payment id, which shouldn't happen.